### PR TITLE
Adding long input for hash precompile; ETH and ERC20 transfer 

### PR DESCRIPTION
**Pull request by cody-wang-cb**
Adding support for generating fixtures for hash precompile with long input, as well as ETH/ERC20 transfer

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -24,12 +24,12 @@ enclave := "devnet"
...
 
-name := "Writer"
+name := "Precompiler"
 script-file := name + ".s.sol"
 
 # Space-separated list of script arguments
-script-args := "1000000"
-script-signature := "run(" + replace_regex(script-args, "\\S+\\s?", "uint256,") + ")"
+script-args := "2 3000000 true"
+script-signature := "run(" + replace_regex(script-args, "^(\\S+)\\s+(\\S+)\\s+(\\S+)$", "uint256,uint256,bool") + ")"
```
Comment (BrianBland) [inline justfile]: This change should no longer be necessary, now that the `script-signature` automatically conforms to passed in numeric and boolean args

File: justfile [hunk 2]
```diff
@@ -53,17 +53,19 @@ create-devnet:
         --args-file network_params.yaml \
         --enclave {{enclave}}
 
+l1-rpc-url := `kurtosis service inspect devnet el-1-geth-lighthouse | grep -- ' rpc: ' | sed 's/.*-> //'`
```
Comment (BrianBland) [inline justfile]: Is this change needed? I originally moved these into the `generate-fixture` script so that these `kurtosis` calls would only need to be made just-in-time, instead of at the start of any `just` call

File: src/Precompiler.sol [hunk 3]
```diff
@@ -48,56 +79,80 @@ contract Precompiler {
...
-                    exponent.length,
-                    modulus.length,
-                    base,
-                    exponent,
-                    modulus
-                )
-            );
+            count += 1;
+            if (use_long) {
+                bytes memory modulus = abi.encodePacked(hashLongString());
```
Comment (BrianBland) [inline src/Precompiler.sol]: If `hashLongString` were to return `bytes` instead of a `string`, then you would not need to call `abi.encodePacked` on this again.

File: justfile (hunk 2 above)
Comment (cody-wang-cb) [reply justfile]: Oh yeah I think this is because `shell()` doesn't work for me for some reason. What's the set up that can make `shell()` work?

File: justfile [hunk 4]
```diff
@@ -121,34 +129,44 @@ generate-fixture:
...
+        --rollup-url {{ROLLUP_URL}} \
         --rollup-path {{rollup-path}} \
         --genesis-path {{genesis-path}} \
         --output {{fixture-file}} \
         {{verbosity}}
 
-# Runs the given fixture through the op-program
-run-fixture:
-    mkdir -p {{parent_directory(op-program-output)}}
+run-fixture name index gasUse bool:
```
Comment (cody-wang-cb) [inline justfile]: Adding args here so that fixture generation and running the fixture are now independent to each other, can be ran at the same time

Comment (BrianBland) [reply justfile]: It should actually be possible to run an arbitrary fixture file using `just fixture-file=./path/to/fixture.json run-fixture`. I'll update the default logic for deriving the `op-program-output` and `cannon-output` paths to match the `fixture-file` name so you don't need to specify all of the arguments manually

File: justfile (hunk 2 above)
Comment (BrianBland) [reply justfile]: Discussed offline - it appears that `shell()` is a relatively new feature of `just`, and required an upgrade

**Review by BrianBland — APPROVED** [review]
